pkg/log/logger.go

```diff
@@ -0,0 +1,17 @@
+// Package log provides a logger interface for logger libraries
+// so that dex does not depend on any of them directly.
+// It also includes a default implementation using Logrus (used by dex previously).
+package log
+
+// Logger serves as an adapter interface for logger libraries
+// so that dex does not depend on any of them directly.
+type Logger interface {
```

Comment:
How many of these functions are actually used by dex? Can we limit this to only that subset?

Reply:
Currently it includes only those functions that are used by dex. However, as I mentioned in the PR description, I would consider implementing the full set. See my comment bellow.